session.go

```diff
@@ -153,6 +174,10 @@ func (s *Session) interestedIn(c *cid.Cid) bool {
 const provSearchDelay = time.Second * 10
 
 func (s *Session) addActivePeer(p peer.ID) {
+	if s.liveWantsLimit == broadcastLiveWantsLimit {
+		s.liveWantsLimit = targetedLiveWantsLimit
+	}
```

Comment:
Not a fan of using "constant" variables like this... how about a `wantBudget()` function (or something like that) that returns `len(s.liveWants) - ((len(s.activePeers) > 0) ? targetedLiveWantsLimit : broadcastLiveWantsLimit)`.

Reply:
yeah, totally.